### ISSUE TITLE
Use built-in Kubernetes resources as input to Controller

### ISSUE THREAD

**Issue by adam-sandor** [java-operator-sdk/java-operator-sdk#349]
It should be possible to build an Operator that acts on Deployment, Service and other Kubernetes resources instead of a Custom Resource. This would support use-cases where someone wants to enrich the functionality of Kubernetes resources or build some kind of monitoring/logging.

**Comment by scootafew** [comment]
I agree, I have a use-case for which exactly this is required. Is there any plan to support this?

**Comment by csviri** [comment]
@scootafew could you describe your use case pls?

**Comment by jmrodri** [comment]
Reviewed August 5th triage.

**Comment by foal** [comment]
E.g to automatically create Istio CR objects based on `io.fabric8.kubernetes.api.model.networking.v1.Ingress`

**Comment by csviri** [comment]
this is supported in v2

**Comment by foal** [comment]
@csviri Is it in 2.0.0-RC1?

**Comment by csviri** [comment]
yes, should be there, but v 2.0.0 will arrive probably tomorrow, if you can wait a little longer, there are some additional fixes.
